### ISSUE TITLE
`php.lang.security.file-inclusion.file-inclusion` causes false positives

### ISSUE THREAD

**Issue by ianmuscat** [returntocorp/sgrep-rules#2562]
**Describe the bug**
`php.lang.security.file-inclusion.file-inclusion` causes false positives in cases where safe usage of `include|include_once|require|require_once` is assigned to a variable. 

**To Reproduce**
```php
<?php

// ok
require_once 'foo/bar.php';

// false positive
$foo = require_once 'foo/bar.php';
```
```php
<?php

// ok
require_once 'foo/bar.php';

// false positive
$pth = 'foo/bar.php';
require_once $pth;
```

**Expected behavior**
The rule should not match on the above test cases

**Priority**
How important is this to you?
- **P1**: this will block me in the near future

**Comment by 0xDC0DE** [comment]
Thanks for the report, @ianmuscat ! This rule could definitely use some improvements. I have converted it to use taint mode, this should be a much more accurate approach. I have added both your examples as test cases and both will behave as expected once this is released.
